src/experimental/tools/mediaCapabilitiesProber/api/index.ts

```diff
@@ -0,0 +1,164 @@
+   */
+  set LogLevel(level: string) {
+    log.setLevel(level);
+  },
+
+  /**
+   * Get logger level
+   * @returns {string}
+   */
+  get LogLevel() : string {
+    return log.getLevel();
+  },
+
+  /**
+   * Get HDCP status. Evaluates if current equipement support given
+   * HDCP revision.
+   * @param {string}
+   * @returns {Promise}
+   */
+  getStatusForHDCP(hdcp: string) : Promise<string> {
+    if (!hdcp) {
+      return Promise.reject("MediaCapabilitiesProbers >>> Bad Arguments: " +
+        "No HDCP Policy specified.");
+    }
+    const config = {
+      hdcp,
+    };
+    const browserAPIS: IBrowserAPIS[] = [
+      "isTypeSupportedWithFeatures",
+      "getStatusForPolicy",
+    ];
+    return probeMediaConfiguration(config, browserAPIS).then((result) => {
+      switch (result) {
+        case 0:
+          return "NotSupported";
+        case 1:
+          return "Unknown";
+        case 2:
+          return "Supported";
+        default:
+          return "NotSupported";
+      }
+    });
+  },
+
+  /**
+   * Get decoding capabilities from a given video and/or audio
+   * configuration.
+   * @param {Object} mediaConfig
+   * @returns {Promise}
+   */
+  getDecodingCapabilities(
+    mediaConfig: IMediaConfiguration
+  ) : Promise<string> {
+    const config = {
+      type: mediaConfig.type,
+      video: mediaConfig.video,
+      audio: mediaConfig.audio,
+    };
+    const browserAPIS: IBrowserAPIS[] = [
+      "isTypeSupported",
+      "isTypeSupportedWithFeatures",
+      "decodingInfos",
+    ];
+    return probeMediaConfiguration(config, browserAPIS).then((result) => {
+      switch (result) {
+        case 0:
+          return "NotSupported";
+        case 1:
+          return "MaybeSupported";
+        case 2:
+          return "Supported";
+        default:
+          return "NotSupported";
+      }
+    });
+  },
+
+  /**
+   * Tells if browser support deciphering with given drm type and configuration.
+   * @param {string} type
+   * @param {Object} drmConfig
+   * @returns {Promise}
+   */
+  isDRMSupported(
+    type: string,
+    drmConfig: IDRMConfiguration
+  ) : Promise<boolean> {
+    const config = {
+      keySystem: {
+        type,
+        configuration: drmConfig,
+      },
+    };
+    const browserAPIS: IBrowserAPIS[] = ["requestMediaKeySystemAccess"];
+    return probeMediaConfiguration(config, browserAPIS).then((result) => {
+      return result === 2;
+    });
+  },
+
+  /**
+   * Get display capabilites. Tells if display can output
+   * with specific video and/or audio constrains.
+   * @param {Object} displayConfig
+   * @returns {Promise}
+   */
+  getDisplayCapabilities(
+    displayConfig: IDisplayConfiguration
+  ) : Promise<string> {
+    const config = { display: displayConfig };
+    const browserAPIS: IBrowserAPIS[] = [
+      "matchMedia",
+      "isTypeSupportedWithFeatures",
+    ];
+    return probeMediaConfiguration(config, browserAPIS).then((result) => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 66. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=236&issues=AWRg5q1zI66nRh8Y2omf&open=AWRg5q1zI66nRh8Y2omf)

[ISSUE_KEY]: AWRg5q1zI66nRh8Y2omf